### PR TITLE
[recipe] {feat}: {@register_policy_loss("geo_mean"); Geometric-Mean Policy Optimization} 

### PR DESCRIPTION
**Pull request by MzeroMiko**
### What does this PR do?

> This is the official implementaion of paper [***Geometric-Mean Policy Optimization***](https://arxiv.org/abs/2507.20673).

### Checklist Before Starting

- [x] Search for similar PRs. Paste at least one query link here: ...
- [x] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `megatron`, `sglang`, `vllm`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `hardware`, `deployment`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`
  - If this PR involves multiple modules, separate them with `,` like `[megatron, fsdp, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][fsdp, megatron] feat: dynamic batching`

### Test

> The code has trained for 100 iterations, and is still running.

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

A new policy loss function has been added into "verl/trainer/ppo/core_algos.py"
```python
@register_policy_loss("geo_mean")
def compute_policy_loss_geo_mean(
    old_log_prob: torch.Tensor,
    log_prob: torch.Tensor,
    advantages: torch.Tensor,
    response_mask: torch.Tensor,
    loss_agg_mode: str = "token-mean",
    config: Optional[DictConfig | AlgoConfig] = None,
) -> tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch. Tensor]:
    ...
```

We also added directory "examples/gmpo_trainer" for quick start.

### Design & Code Changes

> see above

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [x] Read the [Contribute Guide](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md).
- [x] Apply [pre-commit checks](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [ ] Add / Update [the documentation](https://github.com/volcengine/verl/tree/main/docs).
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/volcengine/verl/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...
- [ ] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1) in [the `verl` Slack workspace](https://join.slack.com/t/verl-project/shared_invite/zt-3855yhg8g-CTkqXu~hKojPCmo7k_yXTQ). (If not accessible, please try [the Feishu group (飞书群)](https://applink.larkoffice.com/client/chat/chatter/add_by_link?link_token=772jd4f1-cd91-441e-a820-498c6614126a).)

### CODE REVIEW

File: examples/gmpo_trainer/run_qwen2_5-7b_math.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
+set -x
```
Comment (vermouth1992) [inline examples/gmpo_trainer/run_qwen2_5-7b_math.sh:1]: Could you add another two scripts that trains dapo dataset on qwen 7b and qwen3 30b moe following dapo recipe
- https://github.com/volcengine/verl/blob/main/recipe/dapo/test_dapo_7b_math.sh
- https://github.com/volcengine/verl/blob/main/recipe/dapo/test_dapo_qwen3_30b_math.sh

You can adjust the hyper-parameters as wish. Thanks.

Comment (MzeroMiko) [reply examples/gmpo_trainer/run_qwen2_5-7b_math.sh:1]: Thank you for your advice. I have added these two scripts, together with the README update. However, we may not be able to train those two scripts due to computational resource limitation. If you encountered any problems training them, feel free to contact me.

File: examples/gmpo_trainer/gmpo_teaser.png
Comment (vermouth1992) [inline examples/gmpo_trainer/gmpo_teaser.png:1]: Could you move this image to another repo or some internet link as it will make verl repo large? Thanks!

Comment (MzeroMiko) [reply examples/gmpo_trainer/gmpo_teaser.png:1]: Sure! The image has been deleted and pasted into README with an internet link now!

**Review by vermouth1992 — APPROVED** [review]

**Comment by vermouth1992** [comment]
Please fix pre-commit according to readme and let's merge it

**Review by vermouth1992 — APPROVED** [review]
